lib/std/unicode.mth

```diff
@@ -0,0 +1,105 @@
+module(std.unicode)
+import(std.prelude)
+import(std.world)
+import(std.str)
+
+||| A unicode codepoint
+struct Codepoint {
+    U32
+    --
+
+    inline {
+        def == [ Codepoint Codepoint -- Bool ] {
+            both:/Codepoint ==
+        }
+        
+        ||| Convert a u32 into a codepoint without verifying it
+        def init-unsafe [ U32 -- Codepoint ] {
+            Codepoint
+        }
+    }
+
+    ||| How many utf-16 words are needed to encode this codepoint (always either
+    ||| 1 or 2)
+    def size-utf-16 [ Codepoint -- USize ] {
+        /Codepoint >Int 0xFFFF > if(2u, 1u) >USize
+    }
+
+    ||| Encode the codepoint and append it to the string
+    def show; [ +Str Codepoint -- +Str ] {
```

Comment:
This is great. But I think this word should actually be called `emit;` or `utf8;` because `show;` is intended more to show the value for debugging purposes.

The actual `show;` definition could be this:

```
def show; [ +Str Codepoint -- +Str ] {
   /Codepoint show; " Codepoint" ;
}
```

`show;`

Reply:
Makes sure!

Reply:
fixed!